extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSource.java

```diff
@@ -166,15 +175,21 @@ private void applyAssumeRole(
       AWSCredentialsProvider awsCredentialsProvider
   )
   {
-    String assumeRoleArn = s3InputSourceConfig.getAssumeRoleArn();
-    if (assumeRoleArn != null) {
+    // Do not run if WebIdentityToken file and assumeRole ARN are detected from the environment variable,
+    // we want the default s3ClientBuilder behavior for ServiceAccount + eks.amazonaws.com/role-arn annotation to work.
+    if (s3InputSourceConfig.isWebIdentityTokenEnvConfigured() && s3InputSourceConfig.isAssumeRoleArnEnvConfigured()) {
```

Comment:
Isnt just this condition required and we can remove all the refactoring from 108:132 ?
This is purely based on boolean logic.

Reply:
@cryptoe To be honest, the previous nested if statements makes it really difficult to understand what-goes-where. If in the future someone needs to add another custom function, it will be hard to do so.